tests/test_serialize.py

```diff
@@ -522,3 +522,36 @@ class MyModel(BaseModel):
         ser = field_serializer('x', json_return_type='str')(partialmethod(func, expected=1234))
 
     assert MyModel(x=1234).model_dump() == {'x': '1,234'}
+
+
+def test_serializer_inheritance_override():
```

Comment:
There is a similar [test](https://github.com/pydantic/pydantic/blob/main/tests/test_serialize.py#L430:L444) from [my previous PR related to the multiple serializers function for a field](https://github.com/pydantic/pydantic/commit/0cd57b3e0c75d9622012de542e5ddc110fe55743). I think this one also should fail. Am I wrong?

Reply:
This is that test, reworked